src/visualize/image_like.jl

```diff
@@ -127,35 +109,114 @@ function play{T}(buffer::Array{T, 2}, video_stream, t)
     return reinterpret(T, buffer, (w,h))
 end
 
-"""
-Takes a 3D image and decides if it is a volume or an animated Image.
-"""
-function _default{T<:Colorant, X}(img::Images.Image{T, 3, X}, s::Style, data::Dict)
-    props = img.properties
-    if haskey(props, "timedim")
-        if haskey(props, "spatialorder")
-            get!(data, :spatialorder, join(props["spatialorder"], ""))
+# If the user is using the new Images, ImageAxes will be loaded
+if isdefined(Images, :ImageAxes)
+    unwrap(img::Images.ImageMeta) = unwrap(data(img))
+    unwrap(img::AxisArray) = unwrap(img.data)
+    unwrap(img::AbstractArray) = img
+
+    GLAbstraction.gl_convert{T}(::Type{T}, img::AbstractArray) = _gl_convert(T, unwrap(img))
+    _gl_convert{T}(::Type{T}, img::Array) = gl_convert(T, img)
+
+    """
+        play(img)
+
+    Turns an Image into a video stream
+    """
+    function play{T}(img::HasAxesArray{T, 3})
+        ax = timeaxis(img)
+        if timeaxis(img) != nothing
+            return const_lift(play, unwrap(img), timedim(img), loop(1:length(ax)))
         end
-        timedim = props["timedim"]
-        video_signal = const_lift(play, img.data, timedim, loop(1:size(img, timedim)))
-        return _default(video_signal, s, data)
-    elseif haskey(props, "pixelspacing")
-        spacing = Vec3f0(map(float, img.properties["pixelspacing"]))
-        pdims   = Vec3f0(size(img))
+        error("Image has no time axis: axes(img) = $(axes(img))")
+    end
+
+    """
+    Takes a 3D image and decides if it is a volume or an animated Image.
+    """
+    function _default{T}(img::HasAxesArray{T,3}, s::Style, data::Dict)
+        # We could do this as a @traitfn, except that those don't
+        # currently mix well with non-trait specialization.
+        if timeaxis(img) != nothing
+            data[:spatialorder] = "yx"
```

Comment:
True. Although I grew to hate `data` in the old Images, it seemed so random when you used it and when you didn't. That might have been more of a problem of other functions/packages, though.

I suppose we could add a fallback `timeaxis` for AbstractArrays. But what should it return for an `Array{T,3}`? I'm inclined towards `nothing`, which is what it currently returns for any `AxisArray` that lacks an axis named `:time`.